sogs/model/message.py

```diff
@@ -0,0 +1,94 @@
+from .. import db
+from ..db import insert_and_get_pk, query
+
+from .user import User
+
+
+class Message:
+    """Class representing a DM between users
+
+    Properties:
+        sender: sender user of the message
+        recip: recipant user of the message
+        data: opaque message data
+        signature: signature of data
+    """
+
+    def __init__(self, row=None, *, sender=None, recip=None, data=None, signature=None):
+        """
+        Constructs a Message from a pre-retrieved row *or* sender recipiant data and signature.
+        """
+        if row is None:
+            if None in (sender, recip, data, signature):
+                raise ValueError("Message() error: no row or data provided")
+            if not all(isinstance(arg, User) for arg in (sender, recip)):
+                raise ValueError("Message() error: sender or recipient was not a User model")
+
+            with db.transaction():
+                row = insert_and_get_pk(
+                    """
+            INSERT INTO inbox (sender, recipient, data, signature)
+            VALUES (:sender, ....)
+            """,
+                    "id",
+                    sender=sender.id,
+                    recipiant=recip.id,
+                    data=data,
+                    signture=signature,
+                ).first()
```

Comment:
This is going to return just the id, not the row.

It seems worth adding a `insert_and_get_row` into db.py that does a `RETURNING *` if `have_returning` and otherwise does a `insert_and_get_pk` followed by selecting the row and returning it.